src/private/fnft__nse_discretization.c

```diff
@@ -383,274 +383,29 @@ INT fnft__nse_discretization_phase_factor_b(const REAL eps_t, const UINT D, REAL
-                i += nskip_per_step;
-            }
-            
-            
-            break;
-        case nse_discretization_CF6_4:
-            q_1 = malloc(D * sizeof(COMPLEX));
-            q_3 = malloc(D * sizeof(COMPLEX));
-            r_1 = malloc(D * sizeof(COMPLEX));
-            r_2 = malloc(D * sizeof(COMPLEX));
-            r_3 = malloc(D * sizeof(COMPLEX));
-            if (q_1 == NULL || q_3 == NULL || r_1 == NULL || r_2 == NULL || r_3 == NULL) {
-                ret_code = E_NOMEM;
-                goto release_mem;
-            }
-            
-            ret_code = misc_resample(D, eps_t, q, -eps_t*nskip_per_step*SQRT(15.0)/10.0, q_1);
-            CHECK_RETCODE(ret_code, release_mem);
-            ret_code = misc_resample(D, eps_t, q, eps_t*nskip_per_step*SQRT(15.0)/10.0, q_3);
-            CHECK_RETCODE(ret_code, release_mem);
-            
-            for (i=0; i < D; i++){
-                r_1[i] = -kappa*CONJ(q_1[i]);
-                r_2[i] = -kappa*CONJ(q[i]);
-                r_3[i] = -kappa*CONJ(q_3[i]);
-            }
-            
-            ret_code = nse_discretization_method_weights(&weights,discretization);
-            CHECK_RETCODE(ret_code, release_mem);
-            
-            i = 0;
-            for (isub=0; isub<D_effective; isub=isub+4) {
-                q_preprocessed[isub] = weights[0]*q_1[i]  + weights[1]*q[i] + weights[2]*q_3[i];
-                r_preprocessed[isub] = weights[0]*r_1[i]  + weights[1]*r_2[i] + weights[2]*r_3[i];
-                q_preprocessed[isub+1] = weights[3]*q_1[i]+ weights[4]*q[i] + weights[5]*q_3[i];
-                r_preprocessed[isub+1] = weights[3]*r_1[i]+ weights[4]*r_2[i] + weights[5]*r_3[i];
-                q_preprocessed[isub+2] = weights[6]*q_1[i] + weights[7]*q[i] + weights[8]*q_3[i];
-                r_preprocessed[isub+2] = weights[6]*r_1[i] + weights[7]*r_2[i] +  weights[8]*r_3[i];
-                q_preprocessed[isub+3] = weights[9]*q_1[i] + weights[10]*q[i] +  weights[11]*q_3[i];
-                r_preprocessed[isub+3] = weights[9]*r_1[i] + weights[10]*r_2[i] +  weights[11]*r_3[i];
-                i += nskip_per_step;
-            }
-            break;
-        case nse_discretization_ES4:
-        case nse_discretization_TES4:
-            i = 0;
-            for (isub=0; isub<D_effective; isub=isub+3) {
-                q_preprocessed[isub] = q[i];
-                i += nskip_per_step;
-            }
-            
-            REAL eps_t_sub = eps_t*nskip_per_step;
-            REAL eps_t_sub_2 = POW(eps_t_sub,2);        
-            q_preprocessed[1] = (q_preprocessed[3]-0)/(2*eps_t_sub); 
-            q_preprocessed[2] = (q_preprocessed[3]-2*q_preprocessed[0]+0)/eps_t_sub_2;
-            q_preprocessed[D_effective-2] = (0-q_preprocessed[D_effective-6])/(2*eps_t_sub); 
-            q_preprocessed[D_effective-1] = (0-2*q_preprocessed[D_effective-3]+q_preprocessed[D_effective-6])/eps_t_sub_2;
-            
+    COMPLEX (*r_from_q[3])(COMPLEX);
```

Comment:
Please add a note here as to why three r_to_q functions are needed.

Reply:
That requirement is documented here:

https://github.com/FastNFT/FNFT/blob/972761817fdd892af1873b70b36b10457ef8ad5f/include/private/fnft__akns_discretization.h#L179

Reply:
With the documentation in mind, why can we then choose `r_from_q[0]` == `r_from_q[1]` == `r_from_q[2]` in `fnft__nse_discretization_preprocess_signal`? Should they not be different?